### PR TITLE
Use compressed Evergreen updates by default

### PR DESCRIPTION
**Pull request by hlwarriner**
The main change in this PR is to update Evergreen's default behavior so that it requests packages containing compressed Cobalt Core binaries. This is to reflect the change in the 2025 Software requirements. Note that this change will also impact devices that were certified before Cobalt 25, after they've completed an update to Cobalt 25, although there is an exception for devices using the Memory Mapped file feature. The --use_compressed_updates flag is replaced with a --use_uncompressed_upddates flag, although it should not be used in production.

A few related changes to documentation and tests are also included to reflect that compressed binaries are now the default: not only for updates but also as the system image.

b/341064002

Change-Id: I79c052e580cbb28f638a9d09cf7445c0fce8c3a6

### CODE REVIEW

File: chrome/updater/configurator.cc [hunk 1]
```diff
@@ -74,10 +74,22 @@ Configurator::Configurator(network::NetworkModule* network_module)
     user_agent_string_ = network_module->GetUserAgent();
   }
   if (base::CommandLine::ForCurrentProcess()->HasSwitch(
-          browser::switches::kUseCompressedUpdates)) {
-    use_compressed_updates_.store(true);
-  } else {
+          browser::switches::kUseUncompressedUpdates)) {
     use_compressed_updates_.store(false);
+  } else {
```
[the remark this answers is not in the document]
Comment (hlwarriner) [reply chrome/updater/configurator.cc]: Yeah, thanks - done.

File: starboard/doc/evergreen/evergreen_binary_compression.md [hunk 2]
```diff
@@ -41,56 +44,57 @@ This feature is incompatible with the Memory Mapped file feature that is
...
 `--loader_use_mmap_file` should not be set if compression is enabled for any
-installation slots (next section).
+installation slots.
 
-## Enabling Binary Compression
+## Disabling Binary Compression
 
-Separate steps are required in order for a partner to enable compression for a)
+Before proceeding, please note that binary compression should not be disabled in
+production.
```
Comment (yuying-y) [inline starboard/doc/evergreen/evergreen_binary_compression.md:52]: Add:

Compression is required starting from Cobalt 25. The feature of disabling binary compression should only be used for debugging or emergency.

File: starboard/doc/evergreen/evergreen_binary_compression.md [hunk 3]
```diff
@@ -41,56 +44,57 @@ This feature is incompatible with the Memory Mapped file feature that is
...
-take full advantage of the storage reduction.
+image slot but disabled for the writable slots, or vice versa. However, starting
+with Cobalt 25 partners are required to have the feature enabled for all slots.
 
-## Disabling Binary Compression
+## Enabling Binary Compression
 
-The compression feature is turned off by default. Once enabled, though, it can
-be disabled by undoing the steps required for enablement.
+The compression feature is turned on by default. Once disabaled, though, it can
```
Comment (yuying-y) [inline starboard/doc/evergreen/evergreen_binary_compression.md:83]: Typo: disabaled->disabled

**Review by yuying-y — APPROVED** [review]
Lgtm with minor comments.

File: starboard/doc/evergreen/evergreen_binary_compression.md (hunk 2 above)
Comment (hlwarriner) [reply starboard/doc/evergreen/evergreen_binary_compression.md]: Done

File: starboard/doc/evergreen/evergreen_binary_compression.md (hunk 3 above)
Comment (hlwarriner) [reply starboard/doc/evergreen/evergreen_binary_compression.md]: Done, thanks.

**Pull request merged**